kernelci/storage.py

```diff
@@ -21,6 +21,24 @@
 from kernelci import shell_cmd
 
 
+def discover_files(path):
+    """Discover files recustively so they can then be uploaded
+
+    Recursively walk through a file hierarchy and return a dictionary with the
+    file paths and open file objects which can then be passed directly to
+    upload_files().
+
+    *path* is the path to the file hierarchy where to look for files
+    """
+    artifacts = {}
+    for root, _, files in os.walk(path):
+        for fname in files:
+            px = os.path.relpath(root, path)
+            artifacts[os.path.join(px, fname)] = open(
```

Comment:
Does `requests.post()` call care internally about closing the file? If not maybe we should care about closing files in `upload_files()`?

Reply:
Well, they'll get closed when their refcount hits 0.  Maybe it should be clarified that the returned value from `discover_files()` should not be used more than once as the file data would have been consumed.

Reply:
The docstring already says that it should "return a dictionary with the file paths and open file objects".  So the caller of this function has to close the file (or let it close itself when it goes out of scope).